Src/CSharpier.Cli/EditorConfig/EditorConfigParser.cs

```diff
@@ -0,0 +1,84 @@
+using System.IO.Abstractions;
+
+namespace CSharpier.Cli.EditorConfig;
+
+internal static class EditorConfigParser
+{
+    /// <summary>Finds all configs above the given directory as well as within the subtree of this directory</summary>
+    public static List<EditorConfigSections> FindForDirectoryName(
+        string directoryName,
+        IFileSystem fileSystem
+    )
+    {
+        if (directoryName is "")
+        {
+            return new List<EditorConfigSections>();
+        }
+
+        var directoryInfo = fileSystem.DirectoryInfo.FromDirectoryName(directoryName);
+        var editorConfigFiles = directoryInfo
```

Comment:
Thus only a part of the `.editorconfig` of the project is considered. Since the project system / Visual Studio with [`<EditorConfigFiles />`](https://github.com/dailydevops/dotnet-engineering/blob/main/settings/sln.build.targets#L94-L95) allows to load further files. In addition Visual Studios extension the [`Global AnalyzerConfig`](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/configuration-files#global-analyzerconfig).

Reply:
This comes from the doc on AnalyzerConfig, so I don't there need to be any changes to support it.
> global configuration files can't be used to configure editor style settings for IDEs, such as indent size or whether to trim trailing whitespace. Instead, they are designed purely for specifying project-level analyzer configuration options.

I wasn't aware of `EditorConfigFiles` but I'll look into it.

Reply:
It sounds like usage of `EditorConfigFiles` is not recommended and `Global AnalyzerConfig` should be used instead. https://github.com/MicrosoftDocs/visualstudio-docs/issues/6051#issuecomment-1092262347 Which makes it sounds like this is more intended to be used for configuring analyzers.

There is an easy workaround of just adding an `.editorconfig` or `.csahrpierrc` to the root, so it doesn't seem critical to support `EditorConfigFiles`.